k8s/client.go

```diff
@@ -230,13 +231,21 @@ func (client *Client) FindServicesByLabel(ctx context.Context, label, ns string)
 }
 
 func (client *Client) UpdateDeployment(ctx context.Context, name, image, version string) error {
-	deployment, err := client.FindDeploymentByName(ctx, name)
+	deploymentList, err := client.FindDeploymentByLabel(ctx, name)
 	if err != nil {
 		return err
 	}
-	deployment.Spec.Template.Spec.Containers[0].Image = fmt.Sprintf("%s:%s", image, version)
-	_, err = client.AppsV1().Deployments(client.Namespace).Update(ctx, deployment, metav1.UpdateOptions{})
-	return err
+	for _, deployment := range deploymentList.Items {
+		if strings.Contains(deployment.Spec.Template.Spec.Containers[0].Image, "ghcr.io/calyptia/core/calyptia-fluent-bit:") {
+			continue
+		}
```

Comment:
because im searching by label so it can come two types of image `ghcr.io/calyptia/core/calyptia-fluent-bit:` or `ghcr.io/calyptia/core:` so im checking which one before i update the version once im trying to update `core` version

Reply:
@gabrielbussolo mmm, deployments made by the core instance itself, will have the pipelineID label set. Other option is to create a helper method against the "coreDockerImage" const that already exists.

Reply:
@gabrielbussolo I don't think we want references to the calyptia-fluent-bit image, this is internal only. use !coreDockerImage

Reply:
problem is that if i use `coreDockerImage` with contains it will match both once this **_ghcr.io/calyptia/core_/calyptia-fluent-bit** also contains `coreDockerImage`

Reply:
Lets use a proper label selector, instance id + negated label (!non existing label).
```go
LabelSelector: fmt.Sprintf("%s=%s, !%s",
				core.DefaultAggregatorIDLabel, id,
				core.DefaultPipelineIDLabel),
```